cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -216,29 +233,174 @@ def _update_postgres_password(self):
         logger.notice('postgres password successfully updated')
 
-    def _configure(self):
-        files.copy_notice(POSTGRESQL_SERVER)
-        self._init_postgresql_server()
-        enable_remote_connections = \
-            config[POSTGRESQL_SERVER][ENABLE_REMOTE_CONNECTIONS]
-        self._update_configuration(enable_remote_connections)
-        if config[POSTGRESQL_SERVER][POSTGRES_PASSWORD]:
-            self._update_postgres_password()
-
-        systemd.restart(SYSTEMD_SERVICE_NAME, append_prefix=False)
-        systemd.verify_alive(SYSTEMD_SERVICE_NAME, append_prefix=False)
+    def _etcd_is_running(self):
+        status = common.run(['pgrep', 'etcd'], ignore_failures=True)
+        return status.returncode == 0
+
+    def _start_etcd(self):
+        # On the first node, etcd start via systemd will fail because of the
+        # other nodes not being up, so we use this approach instead
+        logger.info('Starting etcd')
+        common.sudo(['systemctl', 'start', 'etcd', '--no-block'])
+        while not self._etcd_is_running():
+            logger.info('Waiting for etcd to start...')
+            time.sleep(1)
+        logger.info('etcd has started')
+
+    def _configure_cluster(self):
+        logger.info('Disabling postgres (will be managed by patroni)')
+        systemd.stop(SYSTEMD_SERVICE_NAME, append_prefix=False)
+        systemd.disable(SYSTEMD_SERVICE_NAME, append_prefix=False)
+
+        logger.info('Deploying cluster certificates')
+        # We need access to the certs, which by default we don't have
+        common.sudo(['chmod', 'a+x', '/var/lib/patroni'])
+        # We currently use the same certificates for etcd, patroni,
+        # and postgres. This should be a reasonable starting approach as
+        # these reside on the same machine and all have the same impact if
+        # compromised (full access to data directly or via injected
+        # configuration changes).
+        for cert_deploy_args in (
+            {
+                'cert_destination': ETCD_SERVER_CERT_PATH,
+                'key_destination': ETCD_SERVER_KEY_PATH,
+                'ca_destination': ETCD_CA_PATH,
+                'owner': ETCD_USER,
+                'group': ETCD_GROUP,
+                'key_perms': '400',
+            },
+            {
+                'cert_destination': PATRONI_REST_CERT_PATH,
+                'key_destination': PATRONI_REST_KEY_PATH,
+                'owner': POSTGRES_USER,
+                'group': POSTGRES_GROUP,
+                'key_perms': '400',
+            },
+            {
+                'cert_destination': PATRONI_DB_CERT_PATH,
+                'key_destination': PATRONI_DB_KEY_PATH,
+                'ca_destination': PATRONI_DB_CA_PATH,
+                'owner': POSTGRES_USER,
+                'group': POSTGRES_GROUP,
+                'key_perms': '400',
+            },
+        ):
+            self.use_supplied_certificates(**cert_deploy_args)
```

Comment:
tbh this would be cleaner as just 3 function calls. You save nothing by using a loop with **kwargs since well you have to write out the kwargs either way, just this way is introducing an intermediate step

Reply:
Fair point- when I initially wrote it it was just a list of certs but it mutated into this as I needed to add more info. Will change it now.